group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -1695,8 +1683,8 @@ public void scheduleUnloadOperation(
         scheduleInternalOperation("UnloadCoordinator(tp=" + tp + ", epoch=" + partitionEpoch + ")", tp, () -> {
             CoordinatorContext context = coordinators.get(tp);
             if (context != null) {
+                context.lock.lock();
```

Comment:
is there a semantic reason to move this out of the try block or did we just do it for consistency?

Reply:
I moved it for consistency but it also seems better to lock before entering the try..catch because unlocking if lock fails would be wrong.